terraform/gcp/modules/tuf/kms.tf

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2023 The Sigstore Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_kms_key_ring" "tuf-keyring" {
+  name     = var.tuf_keyring_name
+  location = var.kms_location
+  project  = var.project_id
+}
+
+resource "google_kms_crypto_key" "tuf-key" {
+  name     = var.tuf_key_name
+  key_ring = google_kms_key_ring.tuf-keyring.id
+  purpose  = "ASYMMETRIC_SIGN"
+  version_template {
+    algorithm        = "EC_SIGN_P384_SHA384"
+    protection_level = "SOFTWARE"
+  }
+
+  depends_on = [google_kms_key_ring.tuf-keyring]
+}
+
+resource "google_kms_key_ring_iam_member" "tuf-sa-key-iam" {
+  key_ring_id = google_kms_key_ring.tuf-keyring.id
+  role        = "roles/cloudkms.signerVerifier"
+  member      = format("serviceAccount:%s@%s.iam.gserviceaccount.com", var.tuf_service_account_name, var.project_id)
+  depends_on  = [google_kms_key_ring.tuf-keyring, google_service_account.tuf-sa]
+}
```

Comment:
this looks good to me. 

If you decide to remove the service account variable as suggested in other comment, member here would need changing to something like 
`member = "serviceAccount:${google_service_account.tuf-sa.email}"
`